src/statmech.jl

```diff
@@ -98,6 +98,8 @@ function molecular_equilibrium_equations(absolute_abundances, ionization_energie
         #`residuals!` puts the residuals the system of molecular equilibrium equations in `F`
         #`x` is a vector containing the number density of the neutral species of each element
         function residuals!(F, x)
+            x = abs.(x) #don't allow negative number densities
```

Comment:
If you replaced exp() by abs() for performance reasons, shouldn't you also precompute log(T) ? (I don't know enough about how agressive Julia's optimization is, so maybe it's optimized away).
On a similar topic, isn't it better to work with all these equations in log-space, i.e. always work with log[n]'s rather then [n]'s  ?

Reply:
I haven't verified, but I *think* that happens automatically. That said closures have performance problems, so I'm not positive.  I am avoiding doing a performance overhaul here, because this part of the code is going to have to be seriously modified to get Korg working on cooler stars.  (That is the only reason for not trying a fully log-transformed set of equations as well.)